overdrive.py

```diff
@@ -993,21 +1020,22 @@ def book_info_to_metadata(cls, book, include_bibliographic=True, include_formats
 
                 # Samples become links.
                 if 'samples' in format:
-
-                    if not format['id'] in cls.format_data_for_overdrive_format:
+                    overdrive_name = format['id']
+                    internal_names = list(cls.internal_formats(overdrive_name))
+                    if not internal_names:
                         # Useless to us.
                         continue
-                    content_type, drm_scheme = cls.format_data_for_overdrive_format.get(format['id'])
-                    if Representation.is_media_type(content_type):
-                        for sample_info in format['samples']:
-                            href = sample_info['url']
-                            links.append(
-                                LinkData(
-                                    rel=Hyperlink.SAMPLE,
-                                    href=href,
-                                    media_type=content_type
+                    for content_type, drm_scheme in internal_names:
+                        if Representation.is_media_type(content_type):
+                            for sample_info in format['samples']:
+                                href = sample_info['url']
+                                links.append(
+                                    LinkData(
+                                        rel=Hyperlink.SAMPLE,
+                                        href=href,
+                                        media_type=content_type
+                                    )
```

Comment:
This isn't relevant at this point, but one change that happened since this code was written was the Overdrive samples switched from being served as overdrive-epub-open to being served as overdrive-ebook or overdrive-audiobook.